ethapi/limit.go

```diff
@@ -0,0 +1,91 @@
+package ethapi
+
+import (
+	"bytes"
+	"encoding/json"
+	"errors"
+)
+
+const (
+	// bufferStartSize default buffer start size in bytes
+	bufferStartSize = 10 * 1024
+)
+
+var (
+	// responseSizeLimit limits maximum size of RPC response
+	responseSizeLimit int
+	// ErrMaxResponseSize is returned if size of the RPC call
+	// response is over defined limit
+	ErrMaxResponseSize = errors.New("response size exceeded")
+	// ErrResponseTooLarge is returned when the result buffer cannot be expanded
+	ErrResponseTooLarge = errors.New("response too large")
+	// ErrCannotInitBuffer is returned when initialization of the buffer failed
+	ErrCannotInitBuffer = errors.New("cannot initialize write buffer")
+)
+
+// SetResponseSizeLimit sets maximum size of RPC response in bytes
+func SetResponseSizeLimit(limit int) {
+	responseSizeLimit = limit
+}
+
+// JsonResultBuffer is a bytes buffer for jsonRawMessage result
+type JsonResultBuffer struct {
+	bytes.Buffer
+}
+
+// NewJsonResultBuffer creates new bytes buffer
+func NewJsonResultBuffer() (b *JsonResultBuffer, err error) {
+	defer func() {
+		if recover() != nil {
```

Comment:
Because it tries to increase the buffer size`b.Grow(bufferStartSize)` to `bufferStartSize` not to grow in small increases if the defaul buffer is small, to avoid allocations for growing.